vmngclient/api/templates.py

```diff
@@ -0,0 +1,299 @@
+            self.get(name)
+            raise TemplateExist(name)
+        except TemplateNotFoundError:
+            cli_template = CliTemplate(self.session, device_model, name, description)
+            cli_template.config = config
+            return cli_template.send_to_device()
+
+
+class CliTemplate:
+    """Cli Template."""
+
+    def __init__(self, session: Session, device_model: DeviceModel, name: str, description: str) -> None:
+        self.session = session
+        self.device_model = device_model
+        self.name = name
+        self.description = description
+        self.config: CiscoConfParse = CiscoConfParse([])
+
+    def load(self, id: str) -> None:
+        """Load config from template.
+
+        Args:
+            id (str): The template id from which load config.
+        """
+        endpoint = f"/dataservice/template/device/object/{id}"
+        config = cast(dict, self.session.get_json(endpoint))
+        self.config = CiscoConfParse(config['templateConfiguration'].splitlines())
+
+    def load_running(self, device: DeviceInfo) -> None:
+        """Load running config from device.
+
+        Args:
+            device (DeviceInfo): The device from which load config.
+        """
+        endpoint = f"/dataservice/template/config/running/{device.uuid}"
+        config = cast(dict, self.session.get_json(endpoint))
+        self.config = CiscoConfParse(config['config'].splitlines())
+
+    def send_to_device(self) -> str:
+        """Send CLI template to device.
+
+        Returns:
+            str: Template id.
+        """
+        config_p = "\n".join(self.config.ioscfg)
+        payload = {
+            "templateName": self.name,
+            "templateDescription": self.description,
+            "deviceType": self.device_model.value,
+            "templateConfiguration": config_p,
+            "factoryDefault": False,
+            "configType": "file",
+        }
+        endpoint = "/dataservice/template/device/cli/"
+        response = cast(dict, self.session.post_json(url=endpoint, data=payload))
+        return response['templateId']
+
+    def update(self, id: str) -> None:
+        """Update CLI template.
+
+        Args:
+            id (str): Template id to update.
+
+        Returns:
+            str: Process id.
+        """
+        config_p = "\n".join(self.config.ioscfg)
+        payload = {
+            "templateId": id,
+            "templateName": self.name,
+            "templateDescription": self.description,
+            "deviceType": self.device_model.value,
+            "templateConfiguration": config_p,
+            "factoryDefault": False,
+            "configType": "file",
+            "draftMode": False,
+        }
+        endpoint = f"/dataservice/template/device/{id}"
+        cast(dict, self.session.put_json(url=endpoint, data=payload))
+
+    def add_to_config(self, add_config: CiscoConfParse, add_before: str) -> None:
+        """Adding config to exist config before existing value.
```

Comment:
maybe: `Add config to existing config before provided value` ?

Additionally, do you think it would be necessary to check if `add_before` value is correct or this str is coming from the response and doesn't need to be evaluated?